router/routes.go

```diff
@@ -136,6 +136,11 @@ func (rm *RouteManager) Add(route *Route) error {
 	}
 	route.closer = make(chan bool)
 	route.adapter = adapter
+	//Stop any existing route with this ID:
+	if( rm.routes[route.ID] != nil ){
```

Comment:
@michaelshobbs I did wonder about throwing an error instead of overwriting (see #307 ). In my use case I need to replace the route, ideally without losing any log lines (which could happen if it were stopped through one HTTP call and started it through another), so I did it this way.

Do others have route-replace use-cases / opinions?